bundles/org.openhab.binding.modbus.sunspec/src/main/java/org/openhab/binding/modbus/sunspec/internal/dto/InverterModelBlock.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.sunspec.internal.dto;
+
+import java.util.Optional;
+
+/**
+ * Model for SunSpec compatible inverter data
+ *
+ * @author Nagy Attila Gabor - Initial contribution
+ *
+ */
+public class InverterModelBlock {
+
+    /**
+     * Type of inverter (single phase, split phase, three phase)
+     */
+    public int phaseConfiguration;
+
+    /**
+     * Length of the block in 16bit words
+     */
+    public int length;
+
+    /**
+     * AC Total Current value
+     */
+    public Integer acCurrentTotal;
```

Comment:
Why are some of these fields primitive and other not? Aren't the nullable ones supposed to be Optional?

Reply:
Some of the fields are mandatory per the standard specification, while others are optional. The ones that are primitives are the mandatory fields, the Optionals are the optional.

Please note, that these aren't all the attributes that can be parsed from the datasource but there are the lease 3 times as much. Having a constructor with so many parameters would be really ugly imho. However this DTO is written by the parser class that acts like a factory: it makes sure that every field is correctly initialized.

Reply:
I don't have a problem with the optional ones being represented with an Optional. What I'm trying to understand is what is the difference in this class for int vs Integer and Integer vs Optional<Integer>. I'm trying to figure out where the boxed primitives fit into this.

Reply:
Ok, now I understand your point, and you are right. This was a bit "evolutionary".

I changed everything to classes, so the are also in sync with the optionals.